apps/users/tests/test_views.py

```diff
@@ -518,16 +522,50 @@ def test_ok_redirects(self):
         r = self.client.get(self.url + '?to=/de/firefox/', follow=True)
         self.assert3xx(r, '/de/firefox/')
 
-    def test_bad_redirects(self):
+    def test_bad_redirect_other_domain(self):
+        r = self.client.post(
+            self.url + '?to=https://example.com/this/is/bad',
+            self.data, follow=True)
+        self.assert3xx(r, '/en-US/firefox/')
+
+    def test_bad_redirect_js(self):
+        r = self.client.post(
+            self.url + '?to=javascript:window.alert("xss");',
+            self.data, follow=True)
+        self.assert3xx(r, '/en-US/firefox/')
+
+    def test_double_login_fxa_enabled(self):
+        self.create_switch('fxa-auth', active=True)
         r = self.client.post(self.url, self.data, follow=True)
+        self.assert3xx(r, migrate_path())
+
+        # If you go to the login page when you're already logged in we bounce
+        # you.
+        r = self.client.get(self.url, follow=True)
         self.assert3xx(r, '/en-US/firefox/')
 
-        for redirect in ['http://xx.com',
-                         'data:text/html,<script>window.alert("xss")</script>',
-                         'mailto:test@example.com',
-                         'file:///etc/passwd',
-                         'javascript:window.alert("xss");']:
-            self.assert3xx(r, '/en-US/firefox/')
+    def test_ok_redirects_fxa_enabled(self):
+        self.create_switch('fxa-auth', active=True)
+        r = self.client.post(
+            self.url + '?to=/de/firefox/', self.data, follow=True)
+        self.assert3xx(r, migrate_path('/de/firefox/'))
```

Comment:
maybe this should use something other than the home page as a test because it looks like the home page is a destination for error cases too.